server/message.go

```diff
@@ -0,0 +1,57 @@
+package server
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/coniks-go/utils"
+)
+
+var (
+	ErrorMalformedClientMessage = errors.New("[coniks] Malformed client request")
+	ErrorNameExisted            = errors.New("[coniks] Registering identity is already registered")
+	ErrorInternalServer         = errors.New("[coniks] Internal server error")
+)
+
+const (
+	RegistrationType = "registration"
+)
+
+type Request struct {
+	Type    string
+	Request interface{}
+}
+
+type RegistrationRequest struct {
+	Username               string `json:"username"`
+	Key                    string `json:"key"`
+	KeyType                string `json:"key_type,omitempty"`
+	AllowUnsignedKeychange bool   `json:"allow_unsigned_key_change,omitempty"`
+	AllowPublicLookup      bool   `json:"allow_public_lookup,omitempty"`
+}
+
+type Response interface{}
+
+type ErrorResponse struct {
+	Error string
+}
+
+type RegistrationResponse struct {
+	TempraryBinding string `json:"tb"`
+}
+
+func NewErrorResponse(err error) *ErrorResponse {
+	return &ErrorResponse{Error: err.Error()}
+}
+
+func (reg *RegistrationRequest) serialize() []byte {
+	// serialize scheme:
```

Comment:
Right, though the policies should be included in the tree. I was thinking the client could use the first few bytes of the user data to encode these flags? Or use the first byte as a bitmap? This way, they're stored in the tree, and the server still doesn't know exactly what's in the user's data blob.